instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/bedrock.py

```diff
@@ -140,6 +218,111 @@ def on_success(self, span: Span, result: dict[str, Any]):
                 [stop_reason],
             )
 
+    def _invoke_model_on_success(
+        self, span: Span, result: dict[str, Any], model_id: str
+    ):
+        original_body = None
+        try:
+            original_body = result["body"]
+            body_content = original_body.read()
+
+            # Use one stream for telemetry
+            stream = io.BytesIO(body_content)
+            telemetry_content = stream.read()
+            response_body = json.loads(telemetry_content.decode("utf-8"))
+            if "amazon.titan" in model_id:
+                self._handle_amazon_titan_response(span, response_body)
+            elif "amazon.nova" in model_id:
+                self._handle_amazon_nova_response(span, response_body)
+            elif "anthropic.claude" in model_id:
+                self._handle_anthropic_claude_response(span, response_body)
+            # Replenish stream for downstream application use
+            new_stream = io.BytesIO(body_content)
+            result["body"] = StreamingBody(new_stream, len(body_content))
```

Comment:
In the event of JSONDecodeError, I think the stream would never be replenished

Reply:
Moved the replenish before the decoding